internal/host/hostcommands/download_boot_artifacts_cmd_test.go

```diff
@@ -0,0 +1,88 @@
+package hostcommands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/host/hostutil"
+	"github.com/openshift/assisted-service/internal/versions"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/auth"
+	"gorm.io/gorm"
+)
+
+var _ = Describe("download_boot_artifacts_cmd", func() {
+	var (
+		ctx            = context.Background()
+		ctrl           *gomock.Controller
+		host           models.Host
+		db             *gorm.DB
+		downloadCmd    *downloadBootArtifactsCmd
+		mockVersions   *versions.MockHandler
+		infraEnv       *common.InfraEnv
+		id             strfmt.UUID
+		infraEnvID     strfmt.UUID
+		stepReply      []*models.Step
+		stepErr        error
+		dbName         string
+		imgSvcURL      string
+		hostFSMountDir string
+	)
+
+	BeforeEach(func() {
+		imgSvcURL = common.TestDefaultConfig.BaseDNSDomain
+		db, dbName = common.PrepareTestDB()
+		hostFSMountDir = "/host"
+		ctrl = gomock.NewController(GinkgoT())
+
+		mockVersions = versions.NewMockHandler(ctrl)
+		id = strfmt.UUID(uuid.New().String())
+		infraEnvID = strfmt.UUID(uuid.New().String())
+		host = hostutil.GenerateTestHostWithInfraEnv(id, infraEnvID, models.HostStatusInsufficient, models.HostRoleWorker)
+		host.Inventory = hostutil.GenerateMasterInventory()
+		Expect(db.Create(&host).Error).To(BeNil())
+
+		infraEnv = hostutil.GenerateTestInfraEnv(infraEnvID)
+		infraEnv.CPUArchitecture = *common.TestDefaultConfig.OsImage.CPUArchitecture
+		infraEnv.OpenshiftVersion = *common.TestDefaultConfig.OsImage.OpenshiftVersion
+		Expect(db.Create(infraEnv).Error).To(BeNil())
+	})
+
+	It("happy flow", func() {
```

Comment:
Probably should test some non-happy flows as well.

Reply:
Also it's better to describe what the test is validating, so maybe something like `It("returns a request with the correct content")`

Reply:
Add some unhappy flows and modified their descriptions! Thanks for letting me know!